affinity/cpp-20/d0796r1.md

```diff
@@ -165,94 +206,163 @@ There are some additional challenges which we have been investigating but are no
+
+The `affinity_query` class template provides an abstraction for a relative affinity value between two `execution_resource`s, derived from a particular `affinity_operation` and `affinity_metric`.
+
+### `affinity_query` types
+
+    using native_affinity_type = see-below;
+
+*Requires:* `native_affinity_type` is an implementation defined integral type capable of storing a native affinity value.
+
+    using error_type = see-below;
+
+*Requires:* `error_type` is an implementation defined integral type capable of storing the an error code value.
+
+### `affinity_query` constructors
+
+    affinity_query(const execution_resource &, const execution_resource &) noexcept;
+
+### `affinity_query` destructor
+
+    ~affinity_query();
+
+### `affinity_query` operators
+
+    native_affinity_type native_affinity() const noexcept;
+
+*Returns:* Unspecified native affinity value.
+
+### `affinity_query` comparisons
+
+    friend expected<size_t, error_type> operator==(const affinity_query&, const affinity_query&);
+    friend expected<size_t, error_type> operator!=const affinity_query&, const affinity_query&);
+    friend expected<size_t, error_type> operator<(const affinity_query&, const affinity_query&);
+    friend expected<size_t, error_type> operator>(const affinity_query&, const affinity_query&);
+    friend expected<size_t, error_type> operator<=(const affinity_query&, const affinity_query&);
+    friend expected<size_t, error_type> operator>=(const affinity_query&, const affinity_query&);
+
+*Returns:* An `expected<size_t, error_type>` where,
+* if the affinity query was succesful, the value of type `size_t` represents the magnitude of the relative affinity;
+* if the affinity query was not successful, the error is an error of type `error_type` which represents the reason for affinity query failed.
+
+> [*Note:* An affinity query is permitted to fail if affinity between the two execution resources cannot be calculated for any reason, such as the resources are of different vendors or communication between the resources is not possible. *--end note*]
+
+> [*Note:* The comparison operators rely on the availability of the `expected` class template (see [P0323r4: std::expected][p0323r4]), if this does not become available then an alternative error/value construct will be adopted instead. *--end note*]
 
 ## Free functions
 
-	const this_system::execution_resource &resource();
+The free function `this_system::resources` is provided for retrieving the `execution_resource`s which encapsualte the hardware platforms available within the system, these are refered to as the *system level resources*.
+
+	std::vector<execution_resource> resources() noexcept;
+
+*Returns:* A std::vector containing all *system level resources*.
+
+*Requires:* If `resources().size() > 0`, `resources()[0]` be the `execution_resource` corroponding to the current thread of execution. The value returned by `resources()` be the same at any point after the invocation of `main`.
```

Comment:
corroponding -> corresponding
resources.size() must at least be 1, and that must be the resource used for std::thread

Reply:
Done.